core/lib/instrument.c

```diff
@@ -3045,9 +3045,15 @@ custom_memory_shared(bool alloc, void *drcontext, dr_alloc_flags_t flags, size_t
                       TESTALL(DR_ALLOC_NON_HEAP | DR_ALLOC_NON_DR, flags),
                   "dr_custom_alloc: reserve/commit-only are only for non-DR non-heap");
     CLIENT_ASSERT(!TEST(DR_ALLOC_RESERVE_ONLY, flags) ||
+
                       !TEST(DR_ALLOC_COMMIT_ONLY, flags),
                   "dr_custom_alloc: cannot combine reserve-only + commit-only");
 #    endif
+    CLIENT_ASSERT(!TEST(DR_ALLOC_CACHE_REACHABLE, flags) ||
+                      !TESTALL(DR_MEMPROT_WRITE | DR_MEMPROT_EXEC, prot) ||
```

Comment:
The protection bits don't matter here, do they?  Anything reachable will be a problem.

Reply:
That's actually what I said in my comment. I wasn't sure if you wanted to be conservative somehow. Removed!